### ISSUE TITLE
ignore-pass-statements introduces unnecessary pass statements

### ISSUE THREAD

**Issue by yarnabrina** [PyCQA/autoflake#206]
If "--ignore-pass-statements" is used, removed imports are being replaced with `pass`.

```python
>>> import autoflake
>>> 
>>> print(autoflake.__version__)
2.0.0
>>> 
>>> code = """
... import os
... 
... print("hello")
... """
>>> 
>>> fixed_code = autoflake.fix_code(code, ignore_pass_statements=True)
>>> print(fixed_code)

pass

print("hello")

>>> 
```

This looks like a bug.

**Comment by fsouza** [comment]
Yeah, removed imports are always replaced with `pass` and then the `pass` statements are cleared, it's an unfortunate combination. We should definitely do better.

**Comment by wlrd** [comment]
@fsouza do we know why this is happening and is anyone taking a look? If not, I can try to dig deeper into figuring out what is going on here.

**Comment by fsouza** [comment]
Oh, it's just how it works, it's an accident of the design: to guarantee that the code is valid after the removal of the import, autoflake plays safe and puts a `pass` statement (trusting that it can be removed later, but then if `ignore-pass-statement` is set, it's never removed), you can see the comment in the code and where it happens: https://github.com/PyCQA/autoflake/blob/60eeb31082bf5290341e84a008d5c832fb04fd13/autoflake.py#L657-L661

We can reuse the logic to determine whether a pass can be removed to safely remove the import, or have a way to identify which `pass`es were introduced by the removal of the import.

No one is currently working on this, feel free to take a stab!

**Comment by KevinMGranger** [comment]
I intend to take a swing at this.

My approach:

1. Generate a random id (just a random string) for the whole execution
2. Replace unused imports instead with `pass # AUTOFLAKE:$RANDOM_ID`
3. Remove `pass` lines matching the above, even if `ignore-pass-statement` is set.

I think this is good enough for now. If we want to address fancier features like [#234: removing newly-empty `TYPE_CHECKING` blocks](https://github.com/PyCQA/autoflake/issues/234) in the future, it might be better to express each line as either a string or some sort of special `RemovedImport()` sentinel.

(Side note: I think #234 as mentioned above is in the same boat as #52).

**Comment by keriksson-rosenqvist** [comment]
Is there any update on this?
